### PR TITLE
tests: add handlers listen integration tests

### PR DESCRIPTION
**Pull request by resmo**
##### ISSUE TYPE
- Feature Pull Request
##### COMPONENT NAME

Integration tests
##### ANSIBLE VERSION

```
2.2.1
```
##### SUMMARY

add integration tests for handler listen

### CODE REVIEW

**Comment by gundalow** [comment]
@resmo Thanks for improving the tests

What's the purpose of the extra check for `notify_listen_ran_2`, as I read it it's defending the same code path as `notify_listen_ran_1`?

Would a check to make sure a different handler _has not_ been fired a good test?

Out of interest did you recently find that handlers were broken, or were you working on updating the handler code and wanted to ensure you hadn't broken something?

**Comment by resmo** [comment]
@gundalow still WIP, does not do anything useful yet. I'll sort it out.

**Comment by gundalow** [comment]
@resmo Ah, yes. I went into auto review mode as I got assigned this. Looks like a good start.

Feel free to shout on on freenode if you need anything

**Comment by mattclay** [comment]
@resmo The integration tests have been moved. The relevant files are now located here:

https://github.com/ansible/ansible/tree/devel/test/integration/targets/handlers

**Comment by mattclay** [comment]
@resmo Are you still planning on working on this PR?

**Comment by resmo** [comment]
yes, I am working on it. catching up after vacation

**Comment by gundalow** [comment]
Added "WIP"  to the summary, that should stop the bot chasing you. Just remove it from the summary when you are done.

**Comment by resmo** [comment]
Thanks @gundalow added integration tests but relaying on PR #17856 to pass

ready_for_review

**Comment by mattclay** [comment]
@resmo If https://github.com/ansible/ansible/pull/17856 is included in stable-2.2 you'll need to create a separate copy of this PR for stable-2.2 since the tests have moved in the devel branch.

**Comment by resmo** [comment]
@mattclay ok, no big deal. :)

**Comment by resmo** [comment]
@gundalow @mattclay this one is ready, dependent code has been merged and tests passed.

**Comment by mattclay** [comment]
@resmo Looks good, but what happened to test case 2? I only see 1, 3 and 4.

**Comment by resmo** [comment]
test case 2 was testing inexistent handler, I should probably find better names for the tests. :)

**Comment by mattclay** [comment]
@resmo Are you going to make changes to the test names, or do you want to leave as-is?

**Comment by mattclay** [comment]
@resmo Tests are failing due to yamllint errors:

```
2016-12-13 02:04:19 test/integration/targets/handlers/test_handlers_listen.yml
2016-12-13 02:04:19   140:1     error    too many blank lines (1 > 0)  (empty-lines)
```

File: test/integration/targets/handlers/runme.sh
Comment (gundalow) [inline test/integration/targets/handlers/runme.sh]: Do we want the same again, but without `ANSIBLE_ERROR_ON_MISSING_HANDLER=false` and ensure it does fail?

**Review by gundalow — COMMENTED** [review]
Looks good, just one idea for an extended test

File: test/integration/targets/handlers/test_handlers_inexistent_notify.yml
Comment (mattclay) [inline test/integration/targets/handlers/test_handlers_inexistent_notify.yml]: Shouldn't this be `notify` instead of `nonify`?

Comment (resmo) [reply test/integration/targets/handlers/test_handlers_inexistent_notify.yml]: indeed!

File: test/integration/targets/handlers/runme.sh
Comment (resmo) [reply test/integration/targets/handlers/runme.sh]: good point. coming back to this.

**Review by mattclay — CHANGES_REQUESTED** [review]
@resmo It looks like we're just waiting on two minor changes from feedback before this can be merged.

**Comment by resmo** [comment]
requested changes implemented, please review again

**Review by mattclay — APPROVED** [review]

**Comment by mattclay** [comment]
@resmo Merged! Thanks for the tests.

Next time when making changes for a review, it would be easier to see the changes made if you added an additional commit instead of replacing the existing commit(s). When merging we can squash the commits from the GitHub UI.

**Comment by resmo** [comment]
@mattclay sure. Thanks!

**Pull request merged**
